src/main/java/org/spongepowered/mod/mixin/core/event/player/MixinEventPlayerInteractBlock.java

```diff
@@ -79,4 +85,15 @@ public EntityInteractionType getInteractionType() {
         return Optional.fromNullable(new Cause(null, this.entityPlayer, null));
     }
 
+    @Override
+    public Optional<Vector3d> getClickedPosition() {
+        return Optional.of(VecHelper.toVector(this.pos).toDouble());
```

Comment:
Looking at this again, `this.pos` is the block position, not the clicked position.
There is no field in the forge event for the clicked position, so you'll have to do an `@Inject` into `NetHandlerPlayServer.processPlayerBlockPlacement`
Specifically, the packet, `C08PacketPlayerBlockPlacement`, has 3 methods you need.
`getPlacedBlockOffsetX()`
`getPlacedBlockOffsetY()`
`getPlacedBlockOffsetZ()`